tests/unit/test_space.py

```diff
@@ -423,3 +423,71 @@ def test_box_deepcopy() -> None:
     box_copy = copy.deepcopy(box)
     npt.assert_allclose(box.lower, box_copy.lower)
     npt.assert_allclose(box.upper, box_copy.upper)
+
+
+@pytest.mark.parametrize(
+    "lower, upper, stepsizes",
+    [
+        pytest.param([0.0, 1.0], [1.0, 2.0], [0.1, 0.2], id="lists"),
+        pytest.param((0.0, 1.0), (1.0, 2.0), (0.1, 0.2), id="tuples"),
+        pytest.param(range(2), range(1, 3), [0.1, 0.2], id="ranges"),
+    ],
+)
+def test_ordinalsearchspace_converts_sequences_to_float64_tensors(lower, upper, stepsizes) -> None:
+    ordinalsp = OrdinalSearchSpace(lower, upper, stepsizes)
+    assert tf.as_dtype(ordinalsp.lower.dtype) is tf.float64
+    assert tf.as_dtype(ordinalsp.upper.dtype) is tf.float64
+    assert tf.as_dtype(ordinalsp.stepsizes.dtype) is tf.float64
+    npt.assert_array_equal(ordinalsp.lower, [0.0, 1.0])
+    npt.assert_array_equal(ordinalsp.upper, [1.0, 2.0])
+    npt.assert_array_equal(ordinalsp.stepsizes, [0.1, 0.2])
+
+
+@pytest.mark.parametrize(
+    "lower, upper, stepsizes",
+    [
+        pytest.param([0.0, 1.0], [1.0, 2.0], [0.1, 0.2]),
+        pytest.param([0.0, 1.0, 2.0], [1.0, 2.0, 3.0], [0.1, 0.2, 0.3]),
+        pytest.param([1.0, 10.0], [10.0, 100.0], [2, 10]),
+    ],
+)
+def test_ordinalsearchspace_sampling_return_correct_rounded_points(lower, upper, stepsizes):
+    ordinalsp = OrdinalSearchSpace(lower, upper, stepsizes)
+    boxsp = Box(lower, upper)
+    tf.random.set_seed(0)
+    ordinalsample = ordinalsp.sample(5)
+    tf.random.set_seed(0)
+    boxsample = boxsp.sample(5)
+    npt.assert_allclose(ordinalsample, tf.round(boxsample / stepsizes) * stepsizes)
+
+
+@pytest.mark.parametrize(
+    "lower, upper, stepsizes",
+    [
+        pytest.param([0.0, 1.0], [1.0, 2.0], [0.1, 0.2]),
+        pytest.param([0.0, 1.0, 2.0], [1.0, 2.0, 3.0], [0.1, 0.2, 0.3]),
+        pytest.param([1.0, 10.0], [10.0, 100.0], [2, 10]),
+    ],
+)
+def test_ordinalsearchspace_sobol_sampling_return_correct_rounded_points(lower, upper, stepsizes):
+    ordinalsp = OrdinalSearchSpace(lower, upper, stepsizes)
+    box = Box(lower, upper)
+    ordinalsample = ordinalsp.sample_sobol(5, skip=0)
+    boxsample = box.sample_sobol(5, skip=0)
+    npt.assert_allclose(ordinalsample, tf.round(boxsample / stepsizes) * stepsizes)
+
+
+@pytest.mark.parametrize(
+    "lower, upper, stepsizes",
+    [
+        pytest.param([0.0, 1.0], [1.0, 2.0], [0.1, 0.2]),
+        pytest.param([0.0, 1.0, 2.0], [1.0, 2.0, 3.0], [0.1, 0.2, 0.3]),
+        pytest.param([1.0, 10.0], [10.0, 100.0], [2, 10]),
+    ],
+)
+def test_ordinalsearchspace_halton_sampling_return_correct_rounded_points(lower, upper, stepsizes):
+    ordinalsp = OrdinalSearchSpace(lower, upper, stepsizes)
+    box = Box(lower, upper)
+    ordinalsample = ordinalsp.sample_halton(5, seed=0)
+    boxsample = box.sample_halton(5, seed=0)
+    npt.assert_allclose(ordinalsample, tf.round(boxsample / stepsizes) * stepsizes)
```

Comment:
You are missing lots of tests (recreate the ones we have for box)

1.  test_ordinal_raises_if_stepsizes_have_invalid_shape
3. test_ordinal_bounds_and_stepsize_attributes()
4. test_ordinal_contains_point
5. test_ordinal_does_not_contain_point
6. test_ordinal___mul___bounds_are_the_concatenation_of_original_bounds()
7. test_ordinal___mul___raises_if_bounds_have_different_types

Reply:
I'll write these, thanks for listing the test functions!